gph/python/ripser_interface.py

```diff
@@ -56,46 +57,44 @@ def _compute_ph_vr_sparse(I, J, V, N, maxHomDim, thresh=-1, coeff=2,
     return ret
 
 
-def _resolve_symmetry_conflicts(coo):
-    """Given a sparse matrix in COO format, filter out any entry at location
-    (i, j) strictly below the diagonal if the entry at (j, i) is also
-    stored. Return row, column and data information for an upper diagonal
-    COO matrix."""
-    _row, _col, _data = coo.row, coo.col, coo.data
+def _sanitize_coo(row, col, data, only_extract_upper=False):
+    """Given a sparse matrix in COO format, either return its upper triangular
+    portion directly, or filter out any entry at location (i, j) strictly below
+    the diagonal if the entry at (j, i) is also stored."""
 
-    below_diag = _col < _row
+    row_orig, col_orig, data_orig = row, col, data
+
+    # Initialize filtered COO data with information in the upper triangle
+    in_upper_triangle = row_orig <= col_orig
```

Comment:
Now we also retrieve the diagonal, is that right ?

Reply:
I thought this was the case before too?

Reply:
You are right, I did not see that row and column were exchanged, sorry

Reply:
No problem!